### PR TITLE
feat (ai/core): add telemetry to generateObject

### PR DESCRIPTION
**Pull request by lgrammel**
## Tasks

- [x] implementation
   - [x] basic
   - [x] doGenerate for json mode
   - [x] doGenerate for tool mode
- [x] docs
   - [x] generateObject reference
   - [x] telemetry guide
- [x] changeset

### CODE REVIEW

File: content/docs/03-ai-sdk-core/60-telemetry.mdx [hunk 1]
```diff
@@ -84,6 +84,22 @@ const result = await generateText({
   - `ai.prompt.messages`: the messages that were passed into the provider
 - `ai.toolCall`: a tool call that is made as part of the generateText call. See [Tool call spans](#tool-call-spans) for more details.
 
+### generateObject function
+
+`generateObject` records 2 types of spans:
+
+- `ai.generateObject`: the full length of the generateObject call. It contains 1 or more `ai.generateObject.doGenerate` spans.
```
Comment (nicoalbanese) [inline content/docs/03-ai-sdk-core/60-telemetry.mdx:91]: wrap generateObject in backticks

File: content/docs/03-ai-sdk-core/60-telemetry.mdx [hunk 2]
```diff
@@ -84,6 +84,22 @@ const result = await generateText({
   - `ai.prompt.messages`: the messages that were passed into the provider
 - `ai.toolCall`: a tool call that is made as part of the generateText call. See [Tool call spans](#tool-call-spans) for more details.
 
+### generateObject function
```
Comment (nicoalbanese) [inline content/docs/03-ai-sdk-core/60-telemetry.mdx:87]: missing backticks

File: content/docs/03-ai-sdk-core/60-telemetry.mdx [hunk 3]
```diff
@@ -84,6 +84,22 @@ const result = await generateText({
...
+
+`generateObject` records 2 types of spans:
+
+- `ai.generateObject`: the full length of the generateObject call. It contains 1 or more `ai.generateObject.doGenerate` spans.
+  It contains the [basic span information](#basic-span-information) and the following attributes:
+  - `operation.name`: `ai.generateObject`
+  - `ai.prompt`: the prompt that was used when calling `generateObject`
+  - `ai.settings.mode`: the object generation mode
+- `ai.generateObject.doGenerate`: a provider doGenerate call.
```
Comment (nicoalbanese) [inline content/docs/03-ai-sdk-core/60-telemetry.mdx:96]: does doGenerate need to be in backticks?

**Review by nicoalbanese — APPROVED** [review]

**Comment by lgrammel** [comment]
@nicoalbanese I did not add backticks on purpose, bc some was header content and for the other part, i wanted the reader to be able to focus on the actual telemetry information. the page is already quite overloaded with code snippets: https://sdk.vercel.ai/docs/ai-sdk-core/telemetry

**Pull request merged**